dtm0/dtx.h

```diff
@@ -0,0 +1,99 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2012-2020 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+
+#pragma once
+
+#ifndef __MOTR_DTM0_DTX_H__
+#define __MOTR_DTM0_DTX_H__
+
+#include "dtm0/tx_desc.h" /* m0_dtm0_tx_desc */
+#include "sm/sm.h"        /* m0_sm */
+#include "dtm/dtm.h"      /* m0_dtx */
+
+enum m0_dtm0_dtx_state {
+	/** An empty dtx. */
+	M0_DDS_INIT,
+	/* dtx has a valid tx record. */
+	M0_DDS_INPROGRESS,
```

Comment:
I'm not against this name in an arbitrary sense. Still i _feel_ it doesn't match with EXECUTED, STABLE, DONE, etc. Can we name it like "launched" or "started" or "sent"?

Reply:
If we want to describe it precisely then the name of the state is "request-is-known-but-no-replies-yet". InProgress is a generic alias for this name. We could use Launched/Started/Sent but it has to be a designated state. In this patch, InProgress does not mean something was sent or logged, it just means the transaction has a valid txr (as it is stated in the comment), and it has no idea about replies. If there are any changes to that in the consequent patches then I'll definitely rename it.